function-aws-api-proxy/src/test/groovy/io/micronaut/function/aws/proxy/payload1/Payload1EagerlyInitializeSingletonTest.groovy

```diff
@@ -0,0 +1,21 @@
+package io.micronaut.function.aws.proxy.payload1
+
+import io.micronaut.function.aws.proxy.Clock
+import spock.lang.Specification
+
+import java.time.LocalDateTime
+
+class Payload1EagerlyInitializeSingletonTest extends Specification {
+
+    void "verify singletons are eagerly intialized for Payload v1 function"() {
+        given:
+        ApiGatewayProxyRequestEventFunction handler = new ApiGatewayProxyRequestEventFunction()
+        sleep(5_000)
```

Comment:
Instead of `Clock` and a 5s sleep in the tests, could we have:

```groovy
@Singleton
class SingletonBean {

    static String PACKAGE = null;

    SingletonBean(ApplicationContext ctx) {
        PACKAGE = ctx.environment.packages.find()
    }
}
```

And then this test becomes:

```
class Payload1EagerlyInitializeSingletonTest extends Specification {

    void "verify singletons are eagerly intialized for Payload v1 function"() {
        given:
        ApiGatewayProxyRequestEventFunction handler = new ApiGatewayProxyRequestEventFunction()

        expect:
        SingletonBean.PACKAGE == ApiGatewayProxyRequestEventFunction.packageName

        cleanup:
        handler.close()
    }
}
```